Sources/SimpleCurrencyConverter/ExchangeRatesNetworkManager.swift

```diff
@@ -0,0 +1,112 @@
+import Foundation
+
+public enum ExchangeRatesError: Error {
+    case failedToGetResponse
+    case failedToConfigureURL
+    case invalidResponse
+    case failedToDecodeResponse
+}
+ 
+enum Keys {
+    static let base = "base"
+    static let symbols = "symbols"
+    static let rates = "rates"
+    static let from = "from"
+    static let to = "to"
+    static let amount = "amount"
+    static let result = "result"
+    
+    static let httpMethod = "GET"
+    static let httpHeader = "apikey"
+}
+
+final class ExchangeRatesNetworkManager {
+    
+    private let session: URLSession
+    
+    init(session: URLSession) {
+        self.session = session
+    }
+    
+    func requestExchangeRate(apiKey: String?, url: URL, base: Currency, target: [Currency]) async throws -> Result<[Currency: Double], Error> {
+        do {
+            let querryItems = [URLQueryItem(name: Keys.base, value: base.rawValue),
+                               URLQueryItem(name: Keys.symbols, value: target.map { $0.rawValue }.joined(separator: ", "))]
+            var urlComps = URLComponents(string: url.absoluteString)
+            urlComps?.queryItems = querryItems
+            
+            guard let modifiedURL = urlComps?.url else {
+                return .failure(ExchangeRatesError.failedToConfigureURL)
+            }
+            
+            var request = URLRequest(url: modifiedURL)
+            request.httpMethod = Keys.httpMethod
+            request.setValue(apiKey, forHTTPHeaderField: Keys.httpHeader)
+            
+            let (data, response) = try await session.data(for: request)
+            
+            guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
+                return .failure(ExchangeRatesError.invalidResponse)
+            }
+            
+            if let jsonObject = try? JSONSerialization.jsonObject(with: data),
+               let jsonDict = jsonObject as? [String: Any],
+               let rates = jsonDict[Keys.rates] as? [String: Double] {
+                return .success(ExchangeRatesNetworkManager.transformToCurrencyKeys(ratesDict: rates))
+            } else {
+                return .failure(ExchangeRatesError.failedToDecodeResponse)
+            }
+        } catch {
+            return .failure(ExchangeRatesError.failedToGetResponse)
+        }
+    }
+    
+    func requestConvert(apiKey: String?, url: URL, amount: Double, base: Currency, target: Currency) async throws -> Result<Double, Error> {
```

Comment:
@ts-dzmitry-kurski Same question here: Is `apiKey` mandatory?
If yes, can you pass `apiKey` as non-optional?